src/app/pages/unuploaded-tests/unuploaded-tests.page.ts

```diff
@@ -34,37 +41,29 @@ export class UnuploadedTestsPage implements OnInit {
   pageState: UnunploadedTestsPageState;
   constructor(
     private store$: Store<StoreModel>,
+    private dateTimeProvider: DateTimeProvider,
+    private slotProvider: SlotProvider,
   ) {
   }
 
-  getUnsubmittedTests() {
-    return this.store$.pipe(
-      select(getTests),
-      // get all slot ids regarded as incomplete from 'tests' slice of state older than 1 day
-      select(getIncompleteTestsSlotOlderThanADay),
-      withLatestFrom(
-        this.store$.pipe(
-          select(getJournalState), // grab 'journal' slice
-        ),
-      ),
-      // filter journal slots by incomplete slot ids inside tests
-      map(([slotIDs, journal]) => getJournalSlotsBySlotIDs(journal, slotIDs)),
-      map((slotItems) =>
-        slotItems.sort((a, b) =>
-          new Date(a.slotData.slotDetail.start).getTime() - new Date(b.slotData.slotDetail.start).getTime())),
-    );
-  }
-
   ngOnInit() {
     this.pageState = {
       appVersion$: this.store$.select(selectVersionNumber),
       employeeName$: this.store$.select(selectEmployeeName),
       employeeId$: this.store$.select(selectEmployeeId).pipe(map(this.getEmployeeNumberDisplayValue)),
       role$: this.store$.select(selectRole).pipe(map(this.getRoleDisplayValue)),
-      unSubmittedTestSlots$: this.getUnsubmittedTests(),
-      slots$: this.getUnsubmittedTests().pipe(
+      unSubmittedTestSlots$: unsubmittedTestSlotsInDateOrder$(this.store$),
+      unSubmittedTestSlotData$: unsubmittedTestSlotsInDateOrder$(this.store$).pipe(
         map((data) => data.map((slot) => slot.slotData)),
       ),
+      unSubmittedTestSlotsOriginal$: this.store$.pipe(
+        select(getJournalState),
+        withLatestFrom(this.store$.pipe(select(getTests))),
+        map(([journal, tests]) => getIncompleteTests(journal, tests, this.dateTimeProvider.now(), this.slotProvider)),
+        map((slotItems: SlotItem[]) =>
+          slotItems.sort((a, b) =>
+            new Date(a.slotData.slotDetail.start).getTime() - new Date(b.slotData.slotDetail.start).getTime())),
+      ),
```

Comment:
Let's remove `unSubmittedTestSlotsOriginal$` in favour of  `unSubmittedTestSlots$` and update the logic inside of `unsubmittedTestSlotsInDateOrder$`

This `unsubmittedTestSlot$` might then become something like
`unSubmittedTestSlots$: unsubmittedTestSlotsInDateOrder$(this.store$, this.dateTimeProvider, this.slotProvider),`